### PR TITLE
Little fix to RPED

### PR DESCRIPTION
**Pull request by Jalleo**
Basically removes the ability to put single assembly units into the RPED, apart from igniters since no machines use it and it was using a weird proc to cause that anyhow. (Stock parts probably needs rewriting in how all of this is handled)

### CODE REVIEW

**Comment by Jalleo** [comment]
For reference for post freeze this means it is more easily expandable post freeze.

File: code/modules/research/stock_parts.dm [hunk 1]
```diff
@@ -20,9 +20,17 @@ If you create T5+ please take a pass at gene_modder.dm [L40]. Max_values MUST fi
...
 	var/alt_sound = null
+	var/static/list/can_go_in = typecacheof(list(
+		/obj/item/reagent_containers/glass/beaker,
+		/obj/item/device/assembly/igniter,
+		/obj/item/stock_parts,
+		/obj/item/stack/ore/bluespace_crystal))
 
 /obj/item/storage/part_replacer/can_be_inserted(obj/item/W, stop_messages = 0, mob/user)
-	return ..() && W.get_part_rating()
+	var/yes = is_type_in_typecache(W, can_go_in)
```
Comment (Cyberboss) [inline code/modules/research/stock_parts.dm]: = can_go_in[W.type]

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by Jalleo** [comment]
Sure will update tonight or tomorrow.

File: code/modules/research/stock_parts.dm [hunk 2]
```diff
@@ -20,9 +20,17 @@ If you create T5+ please take a pass at gene_modder.dm [L40]. Max_values MUST fi
...
+		/obj/item/device/assembly/igniter,
+		/obj/item/stock_parts,
+		/obj/item/stack/ore/bluespace_crystal))
 
 /obj/item/storage/part_replacer/can_be_inserted(obj/item/W, stop_messages = 0, mob/user)
-	return ..() && W.get_part_rating()
+	var/yes = can_go_in[W.type]
+	if(yes)
+		return ..() && TRUE
+	return FALSE
```
Comment (Jalleo) [inline code/modules/research/stock_parts.dm]: @AnturK you and @optimumtact made sense when talking about the ..() here

**Comment by Jalleo** [comment]
@Cyberboss done

**Comment by Jalleo** [comment]
Okay potentially going to fix the consistency issue left from when console screens were removed and it was all changed to use glass instead. Just discussing some things first about it.

File: code/modules/research/stock_parts.dm [hunk 3]
```diff
@@ -20,9 +20,17 @@ If you create T5+ please take a pass at gene_modder.dm [L40]. Max_values MUST fi
...
+		/obj/item/reagent_containers/glass/beaker,
+		/obj/item/device/assembly/igniter,
+		/obj/item/stock_parts,
+		/obj/item/stack/ore/bluespace_crystal))
 
 /obj/item/storage/part_replacer/can_be_inserted(obj/item/W, stop_messages = 0, mob/user)
-	return ..() && W.get_part_rating()
+	var/yes = can_go_in[W.type]
+	if(yes)
+		return ..() && TRUE
```
Comment (Cyberboss) [inline code/modules/research/stock_parts.dm]: ??

Comment (Jalleo) [reply code/modules/research/stock_parts.dm]: see the parent proc. https://github.com/tgstation/tgstation/blob/master/code/game/objects/items/storage/storage.dm#L272

**Comment by Jalleo** [comment]
@Cyberboss changes are done.

**Pull request merged**